src/server/_params.py

```diff
@@ -460,14 +461,21 @@ def parse_source_signal_sets() -> List[SourceSignalSet]:
 
 def parse_geo_sets() -> List[GeoSet]:
     geo_type = request.values.get("geo_type")
+
     if geo_type:
         # old version
         require_any(request, "geo_value", "geo_values", empty=True)
         geo_values = extract_strings(("geo_values", "geo_value"))
         if len(geo_values) == 1 and geo_values[0] == "*":
             return [GeoSet(geo_type, True)]
+
+        for geo_value in geo_values:
+            if geo_value not in delphi_utils.geomap.GeoMapper().get_geo_values(geo_type):
+                raise ValidationFailedException("invalid geo_value for the requested geo_type")
```

Comment:
This is cool!  A few notes though:

- You should put it in a constructor for `GeoSet` instead ([here](https://github.com/cmu-delphi/delphi-epidata/blob/a6cc05838a19aa5af977df29c360b8bb5a95054a/src/server/_params.py#L53)), and then youl'l get this check for free in the call to `parse_geo_arg()` below (among other places).
- It would be good for the Exception to include which of the specified `geo_value`s is the offender.
- nit: you can call/create `GeoMapper()` outside of the loops so its only done once